### PR TITLE
Add validation test about creating pipeline layout with null bind group layout

### PR DESCRIPTION
**Pull request by Jiawei-Shao**
Issue: #4075

<hr>

**Requirements for PR author:**

- [*] All missing test coverage is tracked with "TODO" or `.unimplemented()`.
- [*] New helpers are `/** documented */` and new helper files are found in `helper_index.txt`.
- [*] Test behaves as expected in a WebGPU implementation. (If not passing, explain above.)
- [*] Test have be tested with compatibility mode validation enabled and behave as expected. (If not passing, explain above.)

**Requirements for [reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md):**

- [ ] Tests are properly located in the test tree.
- [ ] [Test descriptions](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) allow a reader to "read only the test plans and evaluate coverage completeness", and accurately reflect the test code.
- [ ] Tests provide complete coverage (including validation control cases). **Missing coverage MUST be covered by TODOs.**
- [ ] Helpers and types promote readability and maintainability.

When landing this PR, be sure to make any necessary issue status updates.

### CODE REVIEW

**Comment by Jiawei-Shao** [comment]
Hi @jzm-intel and @kainino0x ,

With latest webgpu/types I cannot test `undefined` as a member of `GPUPipelineLayoutDescriptor.bindGroupLayouts` so I only test `null` in this PR. I've raised this issue as https://github.com/gpuweb/types/issues/162.

PTAL, thanks!

File: src/webgpu/api/validation/createPipelineLayout.spec.ts [hunk 1]
```diff
@@ -162,3 +163,119 @@ g.test('bind_group_layouts,device_mismatch')
...
+    `
+  )
+  .paramsSubcasesOnly(u =>
+    u //
+      .combine('bindGroupLayoutCount', [1, 2, 3, 4] as const)
+      .combine('bindGroupLayout0', MaybeNullBindGroupLayoutTypes)
+      .combine('bindGroupLayout1', MaybeNullBindGroupLayoutTypes)
+      .combine('bindGroupLayout2', MaybeNullBindGroupLayoutTypes)
+      .combine('bindGroupLayout3', MaybeNullBindGroupLayoutTypes)
+      .filter(t => {
```
Comment (kainino0x) [inline src/webgpu/api/validation/createPipelineLayout.spec.ts:182]: Maybe:
```ts
// Only test cases where at least one of the bind group layouts is null.
const allBGLs = [p.bindGroupLayout0, p.bindGroupLayout1, p.bindGroupLayout2, p.bindGroupLayout3];
const bgls = allBGLs.slice(0, p.bindGroupLayoutCount);
return bgls.includes('Null');
```

**Review by kainino0x — APPROVED** [review]
Optional nits but feel free to land with or without!

File: src/webgpu/api/validation/createPipelineLayout.spec.ts [hunk 2]
```diff
@@ -162,3 +163,96 @@ g.test('bind_group_layouts,device_mismatch')
       t.device.createPipelineLayout({ bindGroupLayouts: [layout0, layout1] });
     }, mismatched);
   });
+
+const MaybeNullBindGroupLayoutTypes = ['Null', 'Undefined', 'Empty', 'NonEmpty'] as const;
```
Comment (kainino0x) [inline src/webgpu/api/validation/createPipelineLayout.spec.ts:167]: Combinatorially adding `undefined` increases the subcase count from 90 to 310, but I think that's a bit too many. This test is not that slow, but our combinatorial explosions in the CTS overall are quite out of hand so I'd like to be more cautious when we're adding new tests.

The null vs undefined thing is a trivial WebIDL check and we really just need one tiny test that makes sure null and undefined behave the same way.

That's a bit of extra work though. Would we get complete coverage by testing only the cases that have _exactly one_ `Null`/`Undefined`/`Empty`? This reduces the whole thing to just 30 subcases without having to write another test.

See #4107
